types/webpack/index.d.ts

```diff
@@ -1622,7 +1665,22 @@ declare namespace webpack {
     }
 
     class DefinePlugin extends Plugin {
-        constructor(definitions: { [key: string]: any });
+        constructor(definitions: { [key: string]: DefinePlugin.CodeValuePrimitive});
+        static runtimeValue(
+            fn: ({ module }: { module: compilation.Module }) => DefinePlugin.CodeValuePrimitive,
+            fileDependencies?: string[]
+        ): DefinePlugin.RuntimeValue;
+    }
+
+    namespace DefinePlugin {
+        class RuntimeValue {
+            constructor(
+                fn: ({ module }: { module: compilation.Module }) => CodeValuePrimitive,
```

Comment:
`module` should have a `resource` property but doesn't:


```
webpack.config.ts:115:48 - error TS2551: Property 'resource' does not exist on type 'Module'. Did you mean '_source'?

115                     return JSON.stringify(path.basename(module.resource, '.tsx'));
                                                                   ~~~~~~~~

  node_modules/@types/webpack/index.d.ts:800:13
    800             _source: any;
                    ~~~~~~~
    '_source' is declared here.
```

This code has been working for a while